challenge.py

```diff
@@ -0,0 +1,117 @@
+import os
+import re
+import numpy as np
+
+List_calls = "calls.txt"
+List_files = "list_files.txt"
+result_stat = "stat.txt"
+flag_file = 1
+flag_calls = 1
+def result(filename):
+  
+  #local variable declaration
+  deleted = 0 
+  added = 0 
+  region = 0 
+  
+  #Regular expression declaration 
+  regex_FileName = '^diff --git a.+[ ]b/(.+?)\n'
+  regex_Del = "^-(?!--).*"
+  regex_Add = "^\+(?!\+\+).*"
+  regex_region = "^@@ -\d+(?:,\d+)? \+\d+(?:,\d+)?\ @@[ ]?.*"
+  regex_functionCall = "([\w]+)(?=\()"
```

Comment:
This won't catch functions calls. Something like `(?![a-z])[^\:,>,\.]([a-z,A-Z]+[_]*[a-z,A-Z]*)+[(]` would.

Reply:
Hi Mathieu,

I did a comment of my approach,
I tried to use your Regex, but it took so much time in the script without giving result,
Actually, I used my regex using some example here on the following link: https://regex101.com/r/n7B3ox/5
it is catching statement too, but I filter them in the code.

Kind regards,
Amine